testapp/src/main/java/com/google/android/agera/testapp/NotesActivity.java

```diff
@@ -112,21 +125,49 @@ protected void onCreate(final Bundle savedInstanceState) {
     recyclerView.setAdapter(adapter);
     recyclerView.setLayoutManager(new LinearLayoutManager(this));
 
-    final DisplayMetrics displayMetrics = getResources().getDisplayMetrics();
-
     backgroundRepository = repositoryWithInitialValue(Result.<Bitmap>absent())
         .observe()
         .onUpdatesPerLoop()
         .goTo(networkExecutor)
-        .getFrom(() -> "http://www.gravatar.com/avatar/4df6f4fe5976df17deeea19443d4429d?s="
-            + Math.max(displayMetrics.heightPixels, displayMetrics.widthPixels))
-        .transform(url -> httpGetRequest(url).compile())
-        .attemptTransform(httpFunction()).orEnd(Result::failure)
+        .getFrom(() -> URLS)
+        .transform(functionFromListOf(String.class)
+            .map(url -> httpGetRequest(url).compile())
+            .morph(parallelMap(networkExecutor, httpFunction()))
```

Comment:
I wouldn't think this as a logical design. It's very weird to have the downloaded data wait on downloading data before decoding.